flamingo_tools/s3_utils.py

```diff
@@ -0,0 +1,107 @@
+import os
+
+import s3fs
+import zarr
+
+from tqdm import tqdm
+
+# Using incucyte s3 as a temporary measure.
+MOBIE_FOLDER = "/mnt/lustre-emmy-hdd/projects/nim00007/data/moser/lightsheet/mobie"
+SERVICE_ENDPOINT = "https://s3.gwdg.de/"
+BUCKET_NAME = "incucyte-general/lightsheet"
+
+# For MoBIE:
+# https://s3.gwdg.de/incucyte-general/lightsheet
+
+def check_s3_credentials(bucket_name, service_endpoint, credentials):
+    """
+    Check if S3 parameter and credentials were set either as a function input or were exported as environment variables.
+    """
+    if bucket_name is None:
+        bucket_name = os.getenv('BUCKET_NAME')
+        if bucket_name is None:
+            raise ValueError("Provide a bucket name for accessing S3 data.\nEither by using an optional argument or exporting an environment variable:\n--s3_bucket_name <bucket_name>\nexport BUCKET_NAME=<bucket_name>")
+
+    if service_endpoint is None:
+        service_endpoint = os.getenv('SERVICE_ENDPOINT')
+        if service_endpoint is None:
+            raise ValueError("Provide a service endpoint for accessing S3 data.\nEither by using an optional argument or exporting an environment variable:\n--s3_service_endpoint <endpoint>\nexport SERVICE_ENDPOINT=<endpoint>")
+
+    if credentials is None:
```

Comment:
It would be nice to also check if we have a credential file in the default location (`~/.aws/credentials`) and parse it if avaialble.